tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/SetMembershipSimplifier.scala

```diff
@@ -7,9 +7,18 @@ import at.forsyte.apalache.tla.lir.transformations.{LanguageWatchdog, Transforma
 import at.forsyte.apalache.tla.lir.values._
 
 /**
- * A simplifier that rewrites vacuously true membership tests based on type information.
+ * A simplifier that rewrites expressions commonly found in `TypeOK`. Assumes expressions to be well-typed.
  *
- * For example, `x \in BOOLEAN` is rewritten to `TRUE` if x is typed BoolT1.
+ * After Apalache's type-checking, we can rewrite some expressions to simpler forms. For example, the (after
+ * type-checking) vacuously true `x \in BOOLEAN` is rewritten to `TRUE` (as `x` must be a `BoolT1`).
+ *
+ * We currently perform the following simplifications (for applicable sets AS, see [[isApplicable]]):
+ *   - `n \in Nat` ~> `x >= 0`
+ *   - `b \in BOOLEAN`, `i \in Int`, `r \in Real` ~> `TRUE`
+ *   - `seq \in Seq(AS)` ~> `TRUE`
+ *   - `set \in SUBSET AS` ~> `TRUE`
+ *   - `fun \in [AS1 -> AS2]` ~> `TRUE`
+ *   - `fun \in [Dom -> AS]` ~> `DOMAIN fun = Dom`
```

Comment:
Can you add `tup \in S1 \X ... \X SN` ~> `TRUE`? 
Maybe even `tup \in SimplifiableSet1 \X ... \X SimplifiableSetN` ~> ` cond(SimplifiableSet1) /\ ... /\ cond(SimplifiableSetN)`, where `SimplifiableSet` is such that `transformMembership` has a case for it?

Reply:
Tuples, records etc are a TODO in https://github.com/informalsystems/apalache/issues/723

Reply:
Fair enough, ignore this comment then.